cmd/service/metro/component.go

```diff
@@ -24,10 +24,10 @@ func NewComponent(ctx context.Context, component string, cfg config.Config) (*Co
 
 	switch component {
 	case Web:
-		svc = web.NewService(ctx, &cfg.Web)
+		svc = web.NewService(ctx, &cfg.Web, &cfg.Registry)
```

Comment:
should we pass registry config to each component now ? or we should initialize the singleton outside the component

Reply:
`Registry` is inited once as a singleton inside the `NewService` construct and stored in the `service` struct for re-use. This is done once on every componet boot-up.